app/services/price_policy_updater.rb

```diff
@@ -0,0 +1,54 @@
+class PricePolicyUpdater
+  def self.update_all!(price_policies, start_date, expire_date, params)
+    self.new(price_policies, start_date, expire_date, params).update_all!
+  end
+
+  def self.destroy_all_for_product!(product, start_date)
+    price_policies = product.price_policies.for_date(start_date)
+    raise ActiveRecord::RecordNotFound if price_policies.none?
+    self.new(price_policies, start_date).destroy_all!
+  end
+
+  def initialize(price_policies, start_date, expire_date = nil, params = nil)
+    @price_policies = price_policies
+    @start_date = start_date
+    @expire_date = expire_date
+    @params = params
+  end
+
+  def destroy_all!
+    ActiveRecord::Base.transaction do
+      @price_policies.all?(&:destroy) || raise(ActiveRecord::Rollback)
```

Comment:
Is `@price_policies` an array at this point, or is there a reason we can't use `destroy_all`?